packages/nx/src/utils/code-frames.ts

```diff
@@ -103,14 +101,18 @@ function getMarkerLines(
 export function codeFrameColumns(
   rawLines: string,
   loc: NodeLocation,
-  opts: Object = {}
+  opts: {
+    linesAbove?: number;
+    linesBelow?: number;
+    highlightedLines?: string;
+  } = {}
 ): string {
   const defs = getDefs(chalk);
   const lines = rawLines.split(NEWLINE);
   const { start, end, markerLines } = getMarkerLines(loc, lines, opts);
 
   const numberMaxWidth = String(end).length;
-  const highlightedLines = highlight(rawLines);
+  const highlightedLines = opts.highlightedLines ?? rawLines;
```

Comment:
Additionally, I would prefer to keep the old functionality, but have `highlight` be an external callback function, e.g.

```typescript
export function codeFrameColumns(
  rawLines: string,
  loc: NodeLocation,
  opts: {
    linesAbove?: number;
    linesBelow?: number;
    highlight?: (rawLines: string) => string;
  } = {}
): string {
   //...
   const highlightedLines = opts.highlight ? highlight(rawLines) : rawLines;
   //...   
 }
```

The function would be then called:
```typescript
import { highlight } from '../code-frames/highlight';

message = codeFrameColumns(
  code,
  { start: { line, column } },
  { highlight }
);
```